File: 728 - Self Dividing Numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> list[int]:
        res = []
        s = set()
        for i in range(left, right + 1):
            if self.check(i):
                res.append(i)
        return res

    def check(self, n: int) -> bool:
        x = n
        while x > 0:
            x, y = divmod(x, 10)
            if y == 0 or n % y != 0:
                return False
        return True
    

File: 728 - Self Dividing Numbers.cpp
#include <vector>
#include <iostream>

using namespace std;


class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> res = {};
        for (int i = left; i <= right; i++) {
            if (check(i)) res.push_back(i);
        }
        return res;
    }
    bool check(int num) {
        int x = num, y;
        while (x > 0) {
            y = x % 10;
            x = x / 10;
            if (y == 0 || num % y != 0) {
                return false;
            }
        }
        return true;
    }
};